### ISSUE TITLE
Hierarchy view shows classes in wrong order

### ISSUE THREAD

**Issue by jarmosaarikko** [NatLibFi/Skosmos#861]
## At which URL did you encounter the problem?
https://finto.fi/ykl/en/page/33
https://finto.fi/udcs/en/page/025403
and many others.

## What steps will reproduce the problem?
1. Go to the class view
2. See the corresponding hierarchy view on the left
3.

## What is the expected output? What do you see instead?

The hierarchy view should show the classes in **alphabethical** order according to the **notation**.

**Expected order** (Which is shown correctly on the NARROWER CONCEPTS list):
33.03 Legal handbooks
33.04 Bodies of laws and directives. Compilations of judicial cases
33.1 International law
33.2 Public and administrative law
33.3 Civil and commercial law

and 

502/504 Environmental science. Conservation of natural resources. Threats to the environment and protection against them
51 Mathematics
52 Astronomy. Astrophysics. Space research. Geodesy

**Instead we see:**
33.1 International law
33.2 Public and administrative law
33.3 Civil and commercial law
33.03 Legal handbooks
33.04 Bodies of laws and directives. Compilations of judicial cases

and 

51 Mathematics
52 Astronomy. Astrophysics. Space research. Geodesy
...
502/504 Environmental science. Conservation of natural resources. Threats to the environment and protection against them

## What browser did you use? (eg. Firefox, Chrome, Safari, Internet explorer)
Chrome.

This may be related to issue https://github.com/NatLibFi/Skosmos/issues/264

**Comment by kouralex** [comment]
Your example is contradictory.
33.04 is before 33.1 when sorting numerically, but 502/504 is not before 51.

How would you like to proceed?

**Comment by jarmosaarikko** [comment]
Yes, in classifications the listing order of notations with captions is always alphabetical, not numeric. Some 
 classifications mix letters and numbers in their notations.

**Comment by miguelahonen** [comment]
It turned out that a hierarchy view still shows classes in wrong order. The following image and the related code describe the problem. 14 -> 14.0 and 36 -> 36.09 are comparable cases but they are shown incoherently. The data is correct which indicates the problem in the code.  

http://finto.fi/ykl/fi/page/14.01
http://finto.fi/ykl/fi/page/36.09

![image](https://user-images.githubusercontent.com/50104570/129341625-7ff7e1b8-568b-430b-8ddd-995b63dbefac.png)

```
ykl:14
  skos:prefLabel "PSYKOLOGIA"@fi, "PSYKOLOGI"@sv, "PSYCHOLOGY"@en ;
  skos:narrower ykl:14.6, ykl:14.01, ykl:14.4, ykl:14.8, ykl:14.1 ;
  skos:related ykl:20, ykl:38, ykl:79.6, ykl:38.5, ykl:38.8, ykl:59.563, ykl:69.35, ykl:59.21, ykl:36.13, ykl:79.3, ykl:17.3, ykl:59.55, ykl:30.13, ykl:39, ykl:70.1, ykl:79.7, ykl:59.56, ykl:58.9, ykl:34.52, ykl:15.1, ykl:79.1, ykl:79.4, ykl:10, ykl:79.5, ykl:79.2, ykl:56.411 ;
  skos:altLabel "Soveltava psykologia (14)"@fi, "Behaviorismi (14+)"@fi, "Psykologiset koulukunnat (14+)"@fi, "Käyttäytyminen, psykologia (14+)"@fi, "Transsexualism, psykologi (14+)"@sv, "Differentiaalipsykologia (14+)"@fi, "Differentialpsykologi (14+)"@sv, "Behaviorism (14+)"@sv, "Psykologi (14+)"@sv, "Beteendepsykologi (14+)"@sv, "Psykotekniikka (14)"@fi, "Själsliv, psykologi (14+)"@sv, "Psykoteknik (14)"@sv, "Sielunelämä, psykologia (14+)"@fi, "Tillämpad psykologi (14)"@sv, "Psykologiska skolor (14+)"@sv, "Käyttäytymisen psykologia (14+)"@fi, "Beslutsfattning, psykologi (14+)"@sv, "Päätöksenteko, psykologia (14+)"@fi, "Sielutiede, psykologia (14+)"@fi, "Beteende, psykologi (14+)"@sv, "Transsukupuolisuus, psykologia (14+)"@fi, "Psykologia (14+)"@fi ;
  skos:closeMatch hklj:129, hklj:120, hklj:128.2, hklj:123.3, hklj:123, hklj:124, hklj:128 ;
  skos:scopeNote """Hit allmänna verk om psykologi
Hit också allmänna verk om differentialpsykologi, tillämpad psykologi och psykoteknik samt verk som handlar om psykologiska skolor i allmänhet
Tillämpningar av psykologi på olika områden klassificeras enligt område
Om verkt handlar om både psykologi och psykiatri, ges företräde åt klass 14+
Psykologins och de psykologiska skolornas historia i allmänhet 14.09"""@sv, """Tähän psykologian yleisteokset
Tähän myös differentiaalipsykologian, soveltavan psykologian ja psykotekniikan yleisteokset sekä teokset, jotka koskevat psykologian koulukuntia yleensä
Psykologian sovellukset eri aloille luokitetaan alan mukaan
Jos teos koskee sekä psykologiaa että psykiatriaa, annetaan etusija luokalle 14+
Psykologian ja psykologisten koulukuntien yleinen historia 14.09"""@fi ;
  a <http://www.yso.fi/onto/ykl-meta/Concept>, skos:Concept ;
  skos:inScheme ykl: ;
  skos:notation "14" ;
  skos:broader ykl:1 .

----------

ykl:14.01
  skos:related ykl:14.1301, ykl:14.1201, ykl:14.401, ykl:38.5201 ;
  skos:prefLabel "Psykologian teoria, filosofia ja tutkimusmenetelmät"@fi, "Psykologisk teori och filosofi. Psykologiska forskningsmetoder"@sv, "Theory, philosophy and research methods of psychology"@en ;
  skos:altLabel "Lämplighetstest, psykologi (14.01)"@sv, "Test, psykologi (14.01)"@sv, "Kokeellinen psykologia (14.01)"@fi, "Soveltuvuuskokeet, psykologia (14.01)"@fi, "Psykologiset testit (14.01)"@fi, "Psykologiska test (14.01)"@sv, "Testit, psykologia (14.01)"@fi, "Experimentell psykologi (14.01)"@sv, "Psykometrik (14.01)"@sv, "Psykometriikka (14.01)"@fi ;
  a <http://www.yso.fi/onto/ykl-meta/Concept>, skos:Concept ;
  skos:broader ykl:14 ;
  skos:closeMatch hklj:120, hklj:122 ;
  skos:inScheme ykl: ;
  skos:scopeNote """Experimentell psykologi i allmänhet
Psykometri och psykologiska test i allmänhet
Teori, filosofi, forskningsmetoder, experiment, test och mätmetoder inom psykologins specialområden klassificeras enligt område"""@sv, """Kokeellinen psykologia yleensä
Psykometriikka ja psykologiset testit yleensä
Psykologian erikoisalojen teoria, filosofia, tutkimusmenetelmät, kokeet, testit ja mittausmenetelmät luokitetaan alan mukaan"""@fi ;
  skos:notation "14.01" .

*************
*************
*************

ykl:36
  skos:altLabel "Kommunism, ekonomiska system (36+)"@sv, "Blandekonomiskt system, ekonomi (36+)"@sv, "Yhteiskulutus, taloustiede (36+)"@fi, "Marxismi, talousjärjestelmät (36+)"@fi, "Samhällsekonomi (36+)"@sv, "Ekonomiska system (36+)"@sv, "Socialism, ekonomiska system (36+)"@sv, "Mikroekonomia, kansantaloustiede (36+)"@fi, "Kiertotalous, taloustiede (36+)"@fi, "Kapitalism, ekonomiska system (36+)"@sv, "Nationalekonomi (36+)"@sv, "Kommunismi, talousjärjestelmät (36+)"@fi, "Sosialismi, talousjärjestelmät (36+)"@fi, "Kansantaloustiede (36+)"@fi, "Jakamistalous, taloustiede (36+)"@fi, "Makroekonomia, kansantaloustiede (36+)"@fi, "Markkinatalous (36)"@fi, "Taloustiede (36+)"@fi, "Cirkulär ekonomi, ekonomisk vetenskap (36+)"@sv, "Talousjärjestelmät (36+)"@fi, "Mikroekonomi, nationalekonomi (36+)"@sv, "Yhteiskuntatalous (36+)"@fi, "Delningsekonomi, ekonomisk vetenskap (36+)"@sv, "Ekonomisk vetenskap (36+)"@sv, "Sekatalousjärjestelmä, taloustiede (36+)"@fi, "Kapitalismi, talousjärjestelmät (36+)"@fi, "Marknadsekonomi (36)"@sv, "Marxism, ekonomiska system (36+)"@sv, "Makroekonomi, nationalekonomi (36+)"@sv, "Yhteiskuntajärjestelmät, taloustiede (36+)"@fi, "Samhällssystem, ekonomi (36+)"@sv ;
  skos:inScheme ykl: ;
  skos:related ykl:69.1, ykl:30, ykl:33, ykl:69.3, ykl:30.11, ykl:69, ykl:32.01 ;
  skos:narrower ykl:36.5, ykl:36.1, ykl:36.3, ykl:36.6, ykl:36.09, ykl:36.2, ykl:36.4 ;
  skos:notation "36" ;
  skos:scopeNote """Hit allmänna verk om företagsekonomi, nationalekonomi och samhällsekonomi
Hit också allmänna verk om ekonomiska system Kapitalism, socialism och blandekonomier som ekonomiska system
Ekonomiska frågor inom olika områden klassificeras enligt område
Om verket innehåller både nationalekonomi och företagsekonomi, ges företräde åt klass 36+
Allmänna läroböcker i ekonomikunskap 36.07
Ekonomisk historia samt olika länders allmänna ekonomiska historia 36.09"""@sv, """Tähän taloustieteen, kansantaloustieteen ja yhteiskuntatalouden yleisteokset
Tähän myös talousjärjestelmiä koskevat yleisteokset
Kapitalismi, sosialismi ja sekatalousjärjestelmä talousjärjestelminä
Eri alojen taloudelliset kysymykset luokitetaan alan mukaan
Jos teoksessa on sekä kansantalous- että liiketaloustiedettä, annetaan etusija luokalle 36+
Taloustiedon yleiset oppikirjat 36.07
Taloustieteen historia sekä eri maiden yleinen taloushistoria 36.09"""@fi ;
  skos:prefLabel "EKONOMISK VETENSKAP. NATIONALEKONOMI"@sv, "ECONOMICS. NATIONAL ECONOMY"@en, "TALOUSTIEDE. KANSANTALOUSTIEDE"@fi ;
  a <http://www.yso.fi/onto/ykl-meta/Concept>, skos:Concept ;
  skos:closeMatch hklj:366, hklj:360 ;
  skos:broader ykl:3 .

----------

ykl:36.09
  skos:scopeNote "Taloustieteen historia sekä eri maiden yleinen taloushistoria"@fi, "Ekonomisk historia samt olika länders allmänna ekonomiska historia"@sv ;
  skos:altLabel "Kapitalism, ekonomisk historia (36.09)"@sv, "Kolonialismi, taloushistoria (36.09)"@fi, "Industrialisering, ekonomisk historia (36.09)"@sv, "Imperialismi, taloushistoria (36.09)"@fi, "Imperialism, ekonomisk historia (36.09)"@sv, "Kolonialism, ekonomisk historia (36.09)"@sv, "Taloustiede, historia (36.09)"@fi, "Teollistuminen, taloushistoria (36.09)"@fi, "Ekonomisk vetenskap, historia (36.09)"@sv, "Kommunism, ekonomisk historia (36.09)"@sv, "Ekonomisk historia (36.09)"@sv, "Industrialismi, taloushistoria (36.09)"@fi, "Kapitalismi, taloushistoria (36.09)"@fi, "Sosialismi, taloushistoria (36.09)"@fi, "Socialism, ekonomisk historia (36.09)"@sv, "Kommunismi, taloushistoria (36.09)"@fi, "Taloushistoria (36.09)"@fi, "Merkantilismi, taloushistoria (36.09)"@fi, "Merkantilism, ekonomisk historia (36.09)"@sv ;
  skos:closeMatch hklj:369 ;
  skos:related ykl:33, ykl:69, ykl:90, ykl:91, ykl:30.11, ykl:32.01, ykl:92 ;
  skos:inScheme ykl: ;
  skos:prefLabel "Ekonomisk historia"@sv, "Economic history"@en, "Taloushistoria"@fi ;
  a skos:Concept, <http://www.yso.fi/onto/ykl-meta/Concept> ;
  skos:broader ykl:36 ;
  skos:notation "36.09" .


```

**Comment by osma** [comment]
The problem was indeed fixed in PR #887, but the JS changes were not reflected in the `footer.inc` file used to adjust layout in Finto.fi. So it was working correctly in a vanilla Skosmos instance, but not in Finto.

Anyway, the specific problem has now been corrected in `footer.inc` in [this commit](https://github.com/NatLibFi/Finto-data/commit/59e5be65ca913bfda1054ab1d80fa6519ad5c584) (see [this comment](https://github.com/NatLibFi/Skosmos/pull/1205#issuecomment-919956270) for more details) and better sorting of notations has also been implemented in PR #1205, so I'm closing this issue.
